pkg/pillar/zfs/zfs.go

```diff
@@ -60,6 +60,41 @@ func CreateDataset(datasetName string) error {
 	return MountDataset(datasetName)
 }
 
+// CreateSnapshot creates a snapshot of the dataset
+// Returns the name of the created snapshot that can be used to rollback
+func CreateSnapshot(datasetName string) (snapshotName string, err error) {
+	snapshotName = time.Now().Format("20060102-150405")
+	// add milliseconds to snapshot name
+	snapshotName += fmt.Sprintf("%03d", time.Now().Nanosecond()/1000000)
```

Comment:
If you care about the time being correct you should presumably save time.Now() in a local variable since the seconds can roll over between the two calls.

Reply:
The milliseconds part can be useful to understand the order in which the snapshots of different volumes where created. For that purpose, it's not critical that the milliseconds where obtained in a separate call. But I think it will look cleaner, if I get them with one call.. I'll fix it, thanks.

Reply:
Done!